### DISCUSSION TITLE
Example of a theme switcher

### DISCUSSION THREAD

**Discussion by marlonmarcello** [stitchesjs/stitches#852] [Help]
Hi! 👋
I was wondering if there is an example of a project with a theme switcher.
I read through all the docs and I think I understand how the `createTheme` works but the examples only show this:

```js
const { styled, css } = createStitches({
  theme: {
    colors: {
      gray100: 'hsl(206,22%,99%)',
      gray200: 'hsl(206,12%,97%)',
      // ...
    },
  },
  space: {},
  fonts: {},
});

const darkTheme = createTheme({
  colors: {
    gray100: 'hsl(206,8%,12%)',
    gray200: 'hsl(206,7%,14%)',
    // ...
  },
  space: {},
  fonts: {},
});

// and then the example goes on shows how to apply it
<div className={darkTheme}>
  <Box>Content nested in dark theme.</Box>
</div>
```

But for example how do I make it so my `darkMode` theme is used automatically if the user has `prefers-color-scheme: dark`?  The example uses the theme on a `div` but how do I go about making it global by using the `styled` function?
What if I wanted to have a theme switcher and let the user pick a theme how does that work?

As an example, coming from `styled-components` all I had to do was pass the theme to the `ThemeProvider`.
I read the [post about the differences between `stitches` and `styled-components`](https://stitches.dev/blog/migrating-from-styled-components-to-stitches) and in there it says `In Stitches, you can define tokens in the config file and seamlessly consume and access directly in the Style Object.` but it doesn't show how to change between the themes.

**Comment by lucastobrazil** [comment]
This might help - I've been working on a ColorModeProvider and general "pattern" for handling this with stitches. 

Would love some feedback on it. I am building a design system that would be consumed by many different products, so was aiming for a portable way of providing the context / toggle functions "out of the box". it handles more than 2 modes

Please let me know what you think - or any ideas!

[CodeSandbox example](https://codesandbox.io/s/stitches-dark-mode-te4ne?file=/src/App.tsx)

**Reply by felipemaccari** [reply]
that is nice @lucastobrazil!
do you think that all of this still worth it instead of using another alternatives like styled-components?

**Reply by lucastobrazil** [reply]
I do - stitches has many benefits over styled-components. Near zero runtime, using css variables reduces runtime cost, variants API is super powerful too

**Reply by felipemaccari** [reply]
nice, i share the same thoughts with my recent studies about stitches!
thanks for sharing the job!

**Comment by marlonmarcello** [comment]
@lucastobrazil this is awesome, thanks so much!!!!

**Reply by lucastobrazil** [reply]
No worries - let me know how you go. Feedback welcomed!
